src/Event/RequestSubscriber.php

```diff
@@ -62,10 +69,31 @@ protected function getMethodForProvider($providerId)
   public function onRequest(GetResponseEvent $event, $eventName, EventDispatcherInterface $dispatcher)
   {
     $request = $event->getRequest();
-    $modifiers = $this->getModifiers();
+    $uri = $request->getPathInfo();
+    [, $uri] = explode('/', $uri);
+
+    // Eliminate the need to get and iterate through all the modifiers as it is
+    // a very resource intensive process.
+    $provider = Provider::load(self::GROUP_PURL_PROVIDER);
+    if ($provider) {
+      $provider_plugin = $this->provider->getProviderPlugin();
+    }
+    if ($provider_plugin) {
+      $modifier = $provider_plugin->getModifierDataByKey($uri);
```

Comment:
The uri will not always be the key for all providers/modifiers, the providers will need the entire request to make their determination.